docker/docker_test.go

```diff
@@ -1,10 +1,37 @@
 package docker
 
-import "testing"
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"github.com/docker/docker/api/types"
+	"strings"
+	"testing"
+)
 
 func TestExecVersion(t *testing.T) {
 	err := Exec("version")
 	if err != nil {
 		t.Error(err)
 	}
 }
+
+func TestExecPipe(t *testing.T) {
```

Comment:
Can you add some extra tests to cover all the cases in ExecPipe?

Reply:
@adam2k  i added few, it's very tricky. Since we are just running gorotine to read data via channel.
i did lot's of practical tests.

Reply:
@adam2k JFYI:

![image](https://www.evernote.com/l/AMM2bE5ChshM-5Jb7MLIDGMjDZwKris-7vIB/image.png)

Reply:
oh okay, so really it's just the `stdout` errors that don't have test coverage in `ExecPipe`.  I think those are OK to leave untested and I get the other issues with testing the Docker commands.  We can figure out something for those with e2e testing later.